test/langtools/tools/javac/bang/BangTypesCompilationTests.java

```diff
@@ -278,6 +304,127 @@ void test(List<? extends Object!> ls1, List<? extends String> ls3) {
                                 Result.Warning,
                                 "compiler.warn.unchecked.nullness.conversion",
                                 1),
+                        new DiagAndCode(
+                                """
+                                class Test {
+                                    static value class Atom {}
+                                    static class Box<X> {}
+                                    void test(Box<? extends Atom!> t1, Box<Atom> t2) {
+                                        t1 = t2;
+                                    }
+                                }
+                                """,
+                                Result.Warning,
+                                "compiler.warn.unchecked.nullness.conversion",
+                                1),
+
+                        new DiagAndCode(
+                                """
+                                class Wrapper<T> {}
+                                class Test<T> {
+                                    Wrapper<T> newWrapper() { return null; }
+                                    void m() {
+                                        Wrapper<T!> w = newWrapper();
+                                    }
+                                }
+                                """,
+                                Result.Warning,
+                                "compiler.warn.unchecked.nullness.conversion",
+                                1),
+                        new DiagAndCode(
+                                """
+                                import java.util.function.*;
+                                class Test {
+                                    void plot(Function<String, String> f) {}
+                                    void m(Function<String!, String> gradient) {
+                                        plot(gradient);
+                                    }
+                                }
+                                """,
+                                Result.Warning,
+                                "compiler.warn.unchecked.nullness.conversion",
+                                1),
+                        new DiagAndCode(
+                                """
+                                import java.util.function.*;
+                                class Test {
+                                    void plot(Function<String!, String> f) {}
+                                    void m(Function<String, String> gradient) {
+                                        plot(gradient);
+                                    }
+                                }
+                                """,
+                                Result.Warning,
+                                "compiler.warn.unchecked.nullness.conversion",
+                                2),  // this needs to be reviewed, the warning is printed twice, should be once only
```

Comment:
Does this have to do with the fact that the subtyping is executed twice (e.g. once in overload, once in the "check" phase) ?

Reply:
yep I think so

Reply:
for some cases like:
```
import java.util.function.*;
class Test {
    void plot(Function<String!, String> f) {}
    void m(Function<String, String> gradient) {
        plot(gradient);
    }
}
```
it was warning twice, there is a double visit by NullabilityComparator to the type. In method IsSubtype::visitClassType, there was a warning being issued when the arguments were compared and then at the end of the method this invocation:

```
    if (result) {
        nullabilityComparator.reset((t1, t2) -> hasNarrowerNullability(t1, t2)).visit(s, t);
    }
```
was provoking the same warning being issued in the same position. We need this invocation to compare the types but it also compares the arguments thus the double warning. I'm using now a field to mark if we have already warned or not.